state-sync/state-sync-v2/state-sync-driver/src/bootstrapper.rs

```diff
@@ -276,20 +276,28 @@ impl<StorageSyncer: StorageSynchronizerInterface> Bootstrapper<StorageSyncer> {
         self.bootstrapped
     }
 
+    /// Marks bootstrapping as complete and notifies any listeners
+    pub fn bootstrapping_complete(&mut self) -> Result<(), Error> {
+        info!("The node has successfully bootstrapped!");
+        self.bootstrapped = true;
+        self.notify_listeners_of_bootstrapping()
+    }
+
     /// Subscribes the specified channel to bootstrap completion notifications
     pub fn subscribe_to_bootstrap_notifications(
         &mut self,
         bootstrap_notifier_channel: oneshot::Sender<Result<(), Error>>,
-    ) {
+    ) -> Result<(), Error> {
         if self.bootstrap_notifier_channel.is_some() {
             panic!("Only one boostrap subscriber is supported at a time!");
         }
 
-        self.bootstrap_notifier_channel = Some(bootstrap_notifier_channel)
+        self.bootstrap_notifier_channel = Some(bootstrap_notifier_channel);
+        self.notify_listeners_of_bootstrapping()
     }
 
     /// Notifies any listeners if we've now bootstrapped
-    fn notify_if_bootstrapped(&mut self) -> Result<(), Error> {
+    fn notify_listeners_of_bootstrapping(&mut self) -> Result<(), Error> {
```

Comment:
tbh: `notify_if_bootstrapped` seems more obvious, vs the new `notify_listeners_of_bootstrapping` seems more like we already know we're bootstrapped.

Reply:
Thanks @gregnazario! Changed to `notify_listeners_if_bootstrapped` 😄